app/controllers/thread_controller.php

```diff
@@ -12,31 +12,47 @@ public function index()
         $total = Thread::countAll();
         $pages = ceil($total / $threads_per_page);
         $this->set(get_defined_vars());
+
     }   
 
     public function view()                
     {
+        
+        $comments_per_page = 10;
+        $pageOne = 1;
         $thread = Thread::get(Param::get('thread_id'));
-        $comments = $thread->getComments();
+
+        $thread_id = Param::get('thread_id');
+
+        $owner = Thread::getOwnerId($thread_id);
+        $ownerId = $owner["creator_id"];
+
+        $ownerUsername = User::getUsername($ownerId);
+
+        $comments = $thread->getComments();        
+        $page = Param::get("page",$pageOne); 
+
+        $pagination = new SimplePagination($page, $comments_per_page);
+        $comments = Comment::getAll($thread_id, $pagination->start_index - 1, $pagination->count + 1);
+        $pagination->checkLastPage($comments);
+        $total = Comment::countAll($thread_id);
+        $pages = ceil($total / $comments_per_page);
+
+        $threadContents = Thread::getBodyContents((int)$thread_id);
         $this->set(get_defined_vars());
+
+
     }
 
-    public function write()                        
+    public function edit()
     {
         $thread = Thread::get(Param::get('thread_id'));
-        $comment = new Comment;
-        $page = Param::get('page_next', 'write');
+        $page = Param::get('page_next', 'edit');
+
         switch ($page) {
-        case 'write':                    
+        case 'edit':                    
             break;
-        case 'write_end':                
-            $comment->username = Param::get('username');
-            $comment->body = Param::get('body');
-            try {            
-                $thread->write($comment);
-            } catch (ValidationException $e) {                    
-                $page = 'write';
-            }                        
+        case 'edit_end':                        
             break;
```

Comment:
Please avoid using magic strings (Apply to all) :bow:
Similar to magic numbers, you may declare these as constant variables.

```
const EDIT = 'edit';
const EDIT_END = 'edit_end';
...
case self::EDIT:
   break;
case self::EDIT_END:
   ...
```